src/Microsoft/Provider.php

```diff
@@ -41,7 +41,7 @@ protected function getAuthUrl($state)
      */
     protected function getTokenUrl()
     {
-        return 'https://login.microsoftonline.com/common/oauth2/v2.0/token';
+        return 'https://login.microsoftonline.com/'.($this->config['tenant'] ?: 'common').'/oauth2/v2.0/token';
```

Comment:
Could we use a `sprintf` here please

Reply:
@atymic sure! Do you want me to update the `getAuthUrl` method too?

Reply:
Sure feel free while you're there :)